src/plugins/vis_type_vislib/public/components/options/metrics_axes/index.tsx

```diff
@@ -299,8 +299,8 @@ function MetricsAxisOptions(props: ValidationVisOptionsProps<BasicVislibParams>)
   }, [stateParams.seriesParams]);
 
   useEffect(() => {
-    vis.setState({ ...vis.serialize(), type: visType });
-  }, [vis, visType]);
+    setValue('type', visType);
+  }, [setValue, visType]);
```

Comment:
I've looked at the code base, and it seems the only case we dynamically changing the `vis.params.type` is only for showing `Show values on chart` switch box of `Panel settings` tab. 
If so, I could propose get rid of effect at all and just rely on uniq metric type in a vis:
```
const visType = useMemo(() => {
    const types = uniq(stateParams.seriesParams.map(({ type }) => type));
    return types.length === 1 ? types[0] : ChartTypes.HISTOGRAM;
  }, [stateParams.seriesParams]);

return 
        {visType === ChartTypes.HISTOGRAM && (
          <SwitchOption
            label= "Show values on chart"
            ...
          />
        )}
```

but I would kindly ask @ppisljar that it's the only case we need `vis.params.type`.
Maybe we could also get rid of this param?

Reply:
@ppisljar ^^^